rsocket/RSocket.cpp

```diff
@@ -1,12 +1,55 @@
 // Copyright 2004-present Facebook. All Rights Reserved.
 
+#include <folly/io/async/EventBaseManager.h>
+
 #include "rsocket/RSocket.h"
 
 namespace rsocket {
 
-std::unique_ptr<RSocketClient> RSocket::createClient(
-    std::unique_ptr<ConnectionFactory> connectionFactory) {
-  return std::make_unique<RSocketClient>(std::move(connectionFactory));
+folly::Future<std::shared_ptr<RSocketClient>> RSocket::createConnectedClient(
+    std::unique_ptr<ConnectionFactory> connectionFactory,
+    SetupParameters setupParameters,
+    std::shared_ptr<RSocketResponder> responder,
+    std::unique_ptr<KeepaliveTimer> keepaliveTimer,
+    std::shared_ptr<RSocketStats> stats,
+    std::shared_ptr<RSocketNetworkStats> networkStats,
+    std::shared_ptr<ResumeManager> resumeManager,
+    std::shared_ptr<ColdResumeHandler> coldResumeHandler,
+    OnRSocketResume) {
+  auto c = std::shared_ptr<RSocketClient>(new RSocketClient(
+      std::move(connectionFactory),
+      std::move(setupParameters),
+      std::move(responder),
+      std::move(keepaliveTimer),
+      std::move(stats),
+      std::move(networkStats),
+      std::move(resumeManager),
+      std::move(coldResumeHandler)));
+
+  return c->connect().then([c]() mutable { return c; });
+}
+
+folly::Future<std::shared_ptr<RSocketClient>> RSocket::createResumedClient(
+    std::unique_ptr<ConnectionFactory> connectionFactory,
+    SetupParameters setupParameters,
+    std::shared_ptr<ResumeManager> resumeManager,
+    std::shared_ptr<ColdResumeHandler> coldResumeHandler,
+    OnRSocketResume,
```

Comment:
Here the resumption arguments come first.  An application which wants a resumed-client cares about these arguments more than the other arguments (they could possibly leave it to default values)